src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -701,6 +767,48 @@ public virtual int ReadByte()
             return r == 0 ? -1 : oneByteArray[0];
         }
 
+        public void ReadExactly(Span<byte> buffer) =>
+            _ = ReadAtLeastCore(buffer, buffer.Length, throwOnEndOfStream: true);
+
+        public void ReadExactly(byte[] buffer, int offset, int count)
+        {
+            ValidateBufferArguments(buffer, offset, count);
+
+            _ = ReadAtLeastCore(buffer.AsSpan(offset, count), count, throwOnEndOfStream: true);
+        }
+
+        public int ReadAtLeast(Span<byte> buffer, int minimumBytes, bool throwOnEndOfStream = true)
+        {
+            ValidateReadAtLeastArguments(buffer.Length, minimumBytes);
+
+            return ReadAtLeastCore(buffer, minimumBytes, throwOnEndOfStream);
+        }
+
+        // No argument checking is done here. It is up to the caller.
+        private int ReadAtLeastCore(Span<byte> buffer, int minimumBytes, bool throwOnEndOfStream)
+        {
+            int totalRead = 0;
+            do
+            {
+                int read = Read(buffer.Slice(totalRead));
+                if (read == 0)
+                {
+                    if (throwOnEndOfStream)
+                    {
+                        ThrowHelper.ThrowEndOfFileException();
+                    }
+                    else
+                    {
+                        return totalRead;
+                    }
```

Comment:
nit - You could get rid of the `else`.
```suggestion
                    return totalRead;
```

Reply:
I missed this in my current changes, but I'll fix it the next time I need to make changes.